src/node/mod.rs

```diff
@@ -380,3 +380,21 @@ pub fn dev_path(dev: dev_t, ty: NodeType) -> io::Result<PathBuf> {
         ),
     ))
 }
+
+/// Returns a Vector with all DRM Nodes we managed to find. There might be duplicates.
+pub fn drm_get_devices() -> Result<Vec<DrmNode>, io::Error> {
```

Comment:
Could this instead return `impl Iterator<Item = io::Result<DrmNode>>`?

Reply:
I'm not sure what would be the best approach here.
We can have IO Errors only by failing to open/read the `/dev` directory.
For the devices themselves, we will only get errors if they aren't DRM devices, which should be the case for the majority of stuff in `/dev`.

Would iterator improve anything in regards to the error handling?

Also, sorry if my answers are a bit confusing. I'm quite new to Rust, so I might be missing the big-picture on how to better handle errors/iterators.

Reply:
> We can have IO Errors only by failing to open/read the `/dev` directory.

That could be handled with:

```rust
io::Result<impl Iterator<Item = io::Result<DrmNode>>>
```

And as discussed below, `drmGetDevices2()` skips errors while opening sub-devices, so we might as well return an iterator over `DrmNode` directly:

```rust
io::Result<impl Iterator<Item = DrmNode>>
```

> For the devices themselves, we will only get errors if they aren't DRM devices, which should be the case for the majority of stuff in `/dev`.

And as below, on most platforms except FreeBSD we should be reading `/dev/dri`.

> Also, sorry if my answers are a bit confusing. I'm quite new to Rust, so I might be missing the big-picture on how to better handle errors/iterators.

No worries, I think everyone would like to educate new contributors on Rust as well as describing the "Rust style" used in drm-rs (as much as we're still discovering what that's supposed to be).

---

Regarding the `drm_get_devices()` function name (based on `drmGetDevices(2)()`), I'd drop the `drm_` prefix since we're already in the `drm` crate (i.e. a fully qualified call would look like `drm::node::drm_get_devices()` which is somewhat superfluous) and then also drop the `get_` prefix because of the [no `get_` prefix convention](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).